apps/src/storage/dataBrowser/dataVisualizer/VisualizerModal.jsx

```diff
@@ -196,9 +198,39 @@ class VisualizerModal extends React.Component {
     return options.join(', ');
   }
 
+  countCells = memoize((records, tableColumns, chartType) => {
+    if (chartType !== ChartType.CROSS_TAB) {
+      return undefined;
+    } else {
+      return records.reduce((bucket, record) => {
```

Comment:
I'd really rather leave this as `bucket`, since I write _all_ my `reduce` functions with `bucket` as the first argument.

It's just the accumulator used by the reduce function - so it's never actually returned or referenced outside of the function. It's just a known place to stick whatever the appropriate return value is, in whatever form it may need to be (object, array, summation, whatever), and any sort of renaming it to something more significant is going to hide the accumulator as well as attach additional external meaning to it.

We're not getting buckets, we're getting sets which we are _only_ using to get sizes of.  This could be wrapped into other code to re-map it to an object pointing from column -> count (instead of column -> set) but that's just adding an extra loop to save accessing a prop on an object later, so I'm not super in favor of it. It's extra complexity and runtime for minimal benefit.

Reply:
Thanks! And just clarifying that my suggestion was to rename the function `countCells` to `getDataColumnBuckets`. I agree `bucket` makes sense here.

Reply:
Do you mean you use the name "bucket" for the accumulator any time you use a reducer? I see another reduce function that uses the name "values" above :)

Fair enough if you want to keep it mapped column -> set to avoid the extra loop, but I think it'd be clearer if the variable name here reflected that (`valuesByColumn` (?) instead of `countCells`)

Reply:
Wow. So I did. That might be a first. 😆 

I did rename it to `getValuesByColumn`, and the `countedCells` variable to `valuesByColumn`